github/handlers/ExecuteBlockActionHandler.ts

```diff
@@ -287,79 +287,13 @@ export class ExecuteBlockActionHandler {
                 }
                 case ModalsEnum.SHARE_PROFILE_PARAMS : {
                     const profileInteractionData = context.getInteractionData().value;
-                    const datAny = profileInteractionData as any;
+                    const dataAny = profileInteractionData as any;
                     const storeData = {
-                        profileParams : datAny as string[]
+                        profileParams : dataAny as string[]
                     }
-
                     await this.persistence.updateByAssociation(new RocketChatAssociationRecord(RocketChatAssociationModel.MISC, "ProfileShareParam"), storeData);
                     break;
                 }
-                case ModalsEnum.SHARE_PROFILE_EXEC : {
-                    let {user, room} = context.getInteractionData();
-                    const block = this.modify.getCreator().getBlockBuilder();
-                    let accessToken = await getAccessTokenForUser(this.read, user ,this.app.oauth2Config) as IAuthData;
-                    const userProfile = await getBasicUserInfo(this.http, accessToken.token);
-
-                    const idRecord = new RocketChatAssociationRecord(RocketChatAssociationModel.MISC, "ProfileShareParam")
-
-                    const profileData = await this.read.getPersistenceReader().readByAssociation(idRecord);
-
-                    let profileShareParams: string[] = [];
-
-                    if (profileData.length == 0){
-                        profileShareParams = ['username', 'avatar', 'email', 'bio', 'followers', 'following' , 'contributionGraph'];
-                    }
-                    else {
-                        const dat = profileData[0] as {profileParams : string[]};
-                        profileShareParams = dat.profileParams;
-                    }
-
-                    if (profileShareParams.includes('avatar')){
-                        block.addImageBlock({
-                            imageUrl : userProfile.avatar,
-                            altText : "User Info"
-                        })
-                    }
-
-                    profileShareParams.map((value) => {
-                        if (value != 'contributionGraph' && value != 'avatar'){
-                            block.addSectionBlock({
-                                text : block.newPlainTextObject(value),
-                            })
-                            block.addContextBlock({
-                                elements : [
-                                    block.newPlainTextObject(userProfile[value], true),
-                                ]
-                            });
```

Comment:
@VipinDevelops can you please share the specefic reason for this change ?

Reply:
Hey @samad-yar-khan, currently, there are two problems with the way we share profile info:

1. We are using an action button to send the info; hence, the modal remains open even after the info is shared. We should use a submit handler instead.
2. We depend on an external project to fetch the contribution graph since GitHub doesn't provide that API. These external dependencies go down from time to time, leading to a buggy user experience.

Reply:
![image](https://github.com/RocketChat/Apps.Github22/assets/99081689/0ad860d9-715d-41c1-b2e4-5f73fb76c828)